language-server/src/main/java/org/sonarlint/languageserver/SonarLintLanguageServer.java

```diff
@@ -200,15 +204,8 @@ void error(String message, Throwable t) {
 
   @Override
   public CompletableFuture<InitializeResult> initialize(InitializeParams params) {
-    String rootUri = params.getRootUri();
-    // rootURI is null when no folder is open (like opening a single file in VSCode)
-    if (rootUri != null) {
-      try {
-        workspaceRoot = new URI(rootUri);
-      } catch (URISyntaxException e) {
-        throw new IllegalStateException(e);
-      }
-    }
+    workspaceFolders.addAll(parseWorkspaceFolders(params.getWorkspaceFolders(), params.getRootUri()));
+    workspaceFolders.sort(Comparator.reverseOrder());
```

Comment:
The natural sort of String is based on lexical order isn't it? Aren't you expecting to have the longest path first?

Reply:
@henryju I think all URI must be absolute values, the folders and files to analyze as well, and then reverse order  assume that all folder paths will be absolute.